packages/pages/src/dev/server/middleware/sendAppHTML.ts

```diff
@@ -61,7 +63,7 @@ export default async function sendAppHTML(
 
   const clientInjectedServerHtml = getServerTemplateDev(
     clientHydrationString,
-    serverHtml,
+    await getServerHtml(),
```

Comment:
This fixes the issue? Can you add a comment to the code about why we're doing this for future reference? Also, is there a way we could add a test for this?

Reply:
Added comment. Honestly, I'm not entirely sure why this works, but it does. 

I started to look into testing it, I think it's doable but it takes a good bit of setup, possibly creating a test vite+express server. I think I have some higher priority items right now so I stopped but I'm happy to loop back to it if you want.